### ISSUE TITLE
Custom self-hosted label pattern matching not working as expected

### ISSUE THREAD

**Issue by grahame-student** [rhysd/actionlint#444]
We use self-hosted runners that have a consistent naming pattern of `xxx` followed by 3 digits.
I've added a configuration file that adds `xxx[0-9][0-9][0-9]` as a pattern and I've tested the pattern with:
- https://pkg.go.dev/path#Match, using `fmt.Println(path.Match("xxx[0-9][0-9][0-9]", "xxx123"))`
- https://pkg.go.dev/path/filepath#Match, using `fmt.Println(filepath.Match("xxx[0-9][0-9][0-9]", "xxx123"))`
both pass

I can see from the super-linter output that the config file is being detected and has added the pattern to the list of valid labels. However `actionlint` is still flagging workflows that use the defined pattern, e.g. `runs-on: xxx065`

I'm using `actionlint-1.6.26` via `super-linter-6.2.0`.
Appreciate the versions are slightly out of date, I'm working on updating to the latest version but there are some niggles to resolve before making the change.

**Comment by grahame-student** [comment]
Here's what my config file looks like for reference
```yml
---
self-hosted-runner:
  # Labels of self-hosted runner in array of string
  labels:
    - xxx[0-9][0-9][0-9]
```

**Comment by grahame-student** [comment]
We solved our niggles and updated to `super-linter-7.1.0`, which includes a newer version of `actionlint`. The updated version is now working as expected.
